### PR TITLE
Convert event banner component into typescript

### PR DESCRIPTION
**Pull request by carpenterd777**
This is part of a series of pull requests that will be made, attempting to slowly convert the project from JSX to TSX. This one sets up the tsconfig as well.

﻿- install typescript
- add config file
- install webpack plugins
- add create-react-app tsconfig to extend
- port event banner to typescript
- Create local variable rather than extend props

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+  "extends": "@tsconfig/create-react-app/tsconfig.json",
+    "compilerOptions": {
+      "outDir": "./built",
+      "allowJs": true,
+      "target": "es5",
+      "jsx": "react"
```
Comment (dyc3) [inline tsconfig.json:7]: recommend enabling `strictNullChecks`

**Review by dyc3 — CHANGES_REQUESTED** [review]
tests are failing

File: src/components/EventBanner/EventBanner.tsx [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+	desc: string;
+	startDate: dayjs.Dayjs;
+	endDate: dayjs.Dayjs;
+	meetingLink: string;
+	isGbm: boolean;
+	now?: dayjs.Dayjs;
+}
+
+const EventBanner = (props: EventBannerProps): JSX.Element | null => {
+	let now: dayjs.Dayjs = props.now ? props.now : dayjs();
```
Comment (dyc3) [inline src/components/EventBanner/EventBanner.tsx:19]: can this be `props.now ?? dayjs()`? or do we not have that enabled

File: src/components/EventBanner/EventBanner.tsx [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+	let flyer = props.flyerSource ? require(`../../assets/flyers/${props.flyerSource}`) : null;
+	let imageElement = flyer ? (
+		<img
+			loading="lazy"
+			src={flyer.default}
+			alt={
+				props.altText
+					? `${props.altText}`
+					: `Flyer for ${props.title ? `${props.title} ` : ""}event`
```
Comment (dyc3) [inline src/components/EventBanner/EventBanner.tsx:29]: `altText` and `title` are marked as required props. We should either mark these as optional or remove these existence checks.

File: src/components/EventBanner/EventBanner.tsx [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+import * as SEC from "../SEC/lib.js";
+
+interface EventBannerProps {
+	flyerSource: any;
+	altText: string;
+	title: string;
+	desc: string;
+	startDate: dayjs.Dayjs;
+	endDate: dayjs.Dayjs;
+	meetingLink: string;
```
Comment (dyc3) [inline src/components/EventBanner/EventBanner.tsx:13]: definitely make this optional
```suggestion
	meetingLink?: string;
```

**Review by dyc3 — CHANGES_REQUESTED** [review]

File: src/components/EventBanner/EventBanner.tsx (hunk 2 above)
Comment (carpenterd777) [reply src/components/EventBanner/EventBanner.tsx:19]: what does that syntax mean?

Comment (dyc3) [reply src/components/EventBanner/EventBanner.tsx:19]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator

**Review by dyc3 — APPROVED** [review]

**Pull request merged**